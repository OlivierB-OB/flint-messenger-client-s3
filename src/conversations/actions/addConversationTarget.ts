import { IAddConversationTargetAction, ADD_CONVERSATION_TARGET } from '../types';

export function addConversationTarget(conversationId: string, target: string): IAddConversationTargetAction {
  return {
    type: ADD_CONVERSATION_TARGET,
    conversationId,
    target,
  };
}
