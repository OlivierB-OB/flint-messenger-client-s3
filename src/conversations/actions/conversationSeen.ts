import { IConversationSeenAction, CONVERSATION_SEEN } from '../types';

export function conversationSeen(id: string, seenDate: string): IConversationSeenAction {
  return {
    type: CONVERSATION_SEEN,
    id,
    seenDate,
  };
}
