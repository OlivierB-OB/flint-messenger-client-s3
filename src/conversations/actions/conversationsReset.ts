import { IConversationsResetAction, CONVERSATIONS_RESET } from '../types';

export function conversationsReset(): IConversationsResetAction {
  return { type: CONVERSATIONS_RESET };
}
