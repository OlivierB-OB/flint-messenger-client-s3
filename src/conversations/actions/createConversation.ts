import { ICreateConversationAction, CREATE_CONVERSATION } from '../types';

export function createConversation(conversationId: string, targets: string[]): ICreateConversationAction {
  return {
    type: CREATE_CONVERSATION,
    conversationId,
    targets,
  };
}
