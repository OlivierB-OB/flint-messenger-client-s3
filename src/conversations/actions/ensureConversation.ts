import { IEnsureConversationAction, ENSURE_CONVERSATION } from '../types';

export function ensureConversation(
  conversationId: string,
  targets: string[],
  createdAt: string,
): IEnsureConversationAction {
  return {
    type: ENSURE_CONVERSATION,
    conversationId,
    targets,
    createdAt,
  };
}
