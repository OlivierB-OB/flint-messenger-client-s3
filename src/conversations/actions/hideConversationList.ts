import { IToggleConversationListAction, TOGGLE_CONVERSATION_LIST } from '../types';

export function hideConversationList(): IToggleConversationListAction {
  return {
    type: TOGGLE_CONVERSATION_LIST,
    value: false,
  };
}
