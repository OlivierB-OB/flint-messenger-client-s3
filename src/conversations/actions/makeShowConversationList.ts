import { batch } from 'react-redux';
import { Action } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import { action } from '../../utils/action';
import { IAppState } from '../../appReducer';
import { updateDrawerContent } from '../../layout/actions/updateDrawerContent';
import { showDrawer } from '../../layout/actions/showDrawer';

export const makeShowConversationList = action(() => {
  return async (dispatch: ThunkDispatch<IAppState, void, Action>) => {
    batch(() => {
      dispatch(updateDrawerContent('conversations'));
      dispatch(showDrawer());
    });
  };
});
