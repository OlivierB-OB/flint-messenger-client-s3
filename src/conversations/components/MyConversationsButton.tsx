import Badge from '@material-ui/core/Badge';
import IconButton from '@material-ui/core/IconButton';
import Forum from '@material-ui/icons/Forum';
import React from 'react';
import { connect } from 'react-redux';
import { Action } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import { IAppState } from '../../appReducer';
import { makeShowConversationList } from '../actions/makeShowConversationList';

interface IShowConversationsButtonProps {
  unseenMessages: number;
  showConversationList: () => void;
}

export function ShowConversationsButton({ unseenMessages, showConversationList }: IShowConversationsButtonProps) {
  return (
    <IconButton aria-label="contacts" onClick={showConversationList}>
      <Badge badgeContent={unseenMessages} color="error" anchorOrigin={{ vertical: 'top', horizontal: 'left' }}>
        <Forum fontSize="large" />
      </Badge>
    </IconButton>
  );
}

const mapStateToProps = ({ conversations }: IAppState) => ({
  unseenMessages: conversations.unseenMessages,
});

const mapDispatchToProps = (dispatch: ThunkDispatch<IAppState, void, Action>) => ({
  showConversationList: () => dispatch(makeShowConversationList()),
});

export const MyConversationsButton = connect(mapStateToProps, mapDispatchToProps)(ShowConversationsButton);
