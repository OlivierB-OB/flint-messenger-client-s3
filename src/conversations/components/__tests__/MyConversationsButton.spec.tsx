import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import { MyConversationsButton } from '../MyConversationsButton';
import { makeShowConversationList } from '../../actions/makeShowConversationList';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('MyConversationsButton', () => {
  let Connected: any = MyConversationsButton;
  let spyShowConversations: jest.Mock;
  let restoreShowConversations: () => void;

  beforeEach(() => {
    [spyShowConversations, restoreShowConversations] = makeShowConversationList.mock(jest.fn());
  });

  afterEach(() => {
    restoreShowConversations();
  });

  it('should provide showConversationList function to ShowContactsButton', async () => {
    const store = mockStore({});
    const component = shallow(<Connected store={store} />);
    (component.prop('showConversationList') as any)();
    expect(spyShowConversations).toHaveBeenCalledWith();
  });
});
