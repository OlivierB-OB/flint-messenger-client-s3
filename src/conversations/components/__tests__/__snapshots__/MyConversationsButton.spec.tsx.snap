// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyConversationsButton should provide showConversationList function to ShowContactsButton 1`] = `
<ShowConversationsButton
  showConversationList={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  unseenMessages="nb unseen messages"
/>
`;
